sentry/src/main/java/io/sentry/Baggage.java

```diff
@@ -0,0 +1,169 @@
+package io.sentry;
+
+import java.io.UnsupportedEncodingException;
+import java.net.URLDecoder;
+import java.net.URLEncoder;
+import java.nio.charset.StandardCharsets;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.TreeSet;
+import org.jetbrains.annotations.ApiStatus;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+@ApiStatus.Experimental
+public final class Baggage {
+
+  static final @NotNull String CHARSET = StandardCharsets.UTF_8.toString();
+  static final @NotNull Integer MAX_VALUE_LENGTH = 8192;
+
+  final @NotNull Map<String, String> keyValues;
```

Comment:
Do we have to support Metadata on the baggage entries? See https://www.w3.org/TR/baggage/#property

Reply:
Also mentioned here https://opentelemetry.io/docs/reference/specification/baggage/api/#set-value

Reply:
Just to make sure we're on the same page: We'll lose the metadata at the moment. But since we're not using this on server side integrations (e.g. Spring) to pass trough incoming baggage headers it won't make a difference and we can leave it out for now. 

I've created #2085 to keep track of the "for later" TODOs.

Reply:
Yup that sounds good to me